java/client/src/main/java/glide/api/commands/StreamBaseCommands.java

```diff
@@ -1372,4 +1372,270 @@ CompletableFuture<GlideString[]> xclaimJustId(
+
+    /**
+     * Returns information about the stream stored at key <code>key</code>.<br>
+     * To get more detailed information use {@link #xinfoStreamFull(GlideString)} or {@link
+     * #xinfoStreamFull(GlideString, int)}.
+     *
+     * @see <a href="https://valkey.io/commands/xinfo-stream/">valkey.io</a> for details.
+     * @param key The key of the stream.
+     * @return A <code>Map</code> of stream information for the given <code>key</code>. See the
+     *     example for a sample response.
+     * @example
+     *     <pre>{@code
+     * // example of using the API:
+     * Map<GlideString, Object> response = client.xinfoStream(gs("myStream")).get();
+     * // the response contains data in the following format:
+     * Map<GlideString, Object> data = Map.of(
+     *     gs("length"), 4L,
+     *     gs("radix-tree-keys"), 1L,
+     *     gs("radix-tree-nodes"), 2L,
+     *     gs("last-generated-id"), gs("1719877599564-0"),
+     *     gs("max-deleted-entry-id"), gs("0-0"),
+     *     gs("entries-added"), 4L,
+     *     gs("recorded-first-entry-id"), gs("1719710679916-0"),
+     *     gs("groups"), 1L,
+     *     gs("first-entry"), new Object {
+     *         gs("1719710679916-0"),
+     *         new GlideString[] {
+     *             gs("foo"), gs("bar"),
+     *             gs("foo"), gs("bar2"),
+     *             gs("some_field"), gs("some_value")
+     *         }},
+     *     gs("last-entry", Object {
+     *         gs("1719877599564-0"),
+     *         new GlideString[] {
+     *             { gs("e4_f"), gs("e4_v") }
+     *         }}
+     * );
+     * // Stream information for "my_stream". Note that "first-entry" and "last-entry" could both be `null` if stream is empty.
+     * }</pre>
+     */
+    CompletableFuture<Map<GlideString, Object>> xinfoStream(GlideString key);
+
+    /**
+     * Returns verbose information about the stream stored at key <code>key</code>.<br>
+     * The output is limited by first <code>10</code> PEL entries.
+     *
+     * @since Redis 6.0 and above.
+     * @see <a href="https://valkey.io/commands/xinfo-stream/">valkey.io</a> for details.
+     * @param key The key of the stream.
+     * @return A <code>Map</code> of detailed stream information for the given <code>key</code>. See
+     *     the example for a sample response.
+     * @example
+     *     <pre>{@code
+     * // example of using the API:
+     * Map<String, Object> response = client.xinfoStreamFull("myStream").get();
+     * // the response contains data in the following format:
+     * Map<String, Object> data = Map.of(
+     *     "length", 4L,
+     *     "radix-tree-keys", 1L,
+     *     "radix-tree-nodes", 2L,
+     *     "last-generated-id", "1719877599564-0",
+     *     "max-deleted-entry-id", "0-0",
+     *     "entries-added", 4L,
+     *     "recorded-first-entry-id", "1719710679916-0",
+     *     "entries", new Object {
+     *         "1719710679916-0",
```

Comment:
each entry should be in its own inner array, see [here](https://github.com/aws/glide-for-redis/blob/main/python/python/glide/async_commands/core.py#L3628), same for other spots in this file

Reply:
Looks like this still needs to be changed